starter_code/src/movies.js

```diff
@@ -1,15 +1,77 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(array) {
+    if (array.length === 0) return null;
+    let newArray = array.sort((a, b) => {
+        if (a.year - b.year === 0) {
+            if (a.title < b.title) return -1;
+            else return 1;
+        } else return a.year - b.year;
+    });
+    return newArray;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+function howManyMovies(array) {
+    if (array.length === 0) return 0;
+    let newArray = array.filter(element => element.director === "Steven Spielberg" && element.genre.includes("Drama"));
+    return newArray.length;
+}
 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(array) {
+    if (array.length === 0) return null;
+    let newArray = array
+        .map(movie => movie.title)
+        .sort((a, b) => {
+            if (a < b) return -1
+            else return 1
+        })
```

Comment:
you could use 

```
  .sort((a, b) => {
      return a.localeCompare(b);
    });
```
More about localeCompare: (it returns either 1, -1, or 0) exactly what the sort method expects :D 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare